src/core/metadata/qgslayermetadata.cpp

```diff
@@ -425,6 +426,83 @@ bool QgsLayerMetadata::operator==( const QgsLayerMetadata &other )  const
          mExtent == other.mExtent;
 }
 
+bool QgsLayerMetadata::contains( const QString &searchString ) const
+{
+  if ( searchString.isEmpty() )
+  {
+    return true;
```

Comment:
Shouldn't this really be false? I think instead of returning true here, if you want an "empty string = show all" behaviour than that should be handled in the filtering code itself (ie QgsLayerMetadataResultsProxyModel )

Reply:
removed the block